extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -168,20 +168,21 @@ export class Bind {
       user().error(TAG, 'Failed to merge scope.', e);
     }
 
-    if (!opt_skipDigest) {
-      this.setStatePromise_ = this.initializePromise_.then(() => {
-        user().fine(TAG, 'State updated; re-evaluating expressions...');
-        return this.digest_();
-      });
-      if (getMode().test) {
-        this.setStatePromise_.then(() => {
-          this.dispatchEventForTesting_('amp:bind:setState');
-        });
-      }
-      return this.setStatePromise_;
-    } else {
+    if (opt_skipEval) {
       return Promise.resolve();
     }
+
+    const promise = this.initializePromise_.then(() => {
+      user().fine(TAG, 'State updated; re-evaluating expressions...');
+      return this.evaluate_().then(results =>
```

Comment:
How about:

```js
    const promise = this.initializePromise_.then(() => {
      user().fine(TAG, 'State updated; re-evaluating expressions...');
      return this.evaluate_();
    }).then(results =>
      this.apply_(results, opt_fromAmpState));
    });
```

to avoid rightward drift?